src/domain/participants/participants.js

```diff
@@ -25,54 +25,58 @@
   try {
     Logger.info('getParticipantsByTypeAndID::begin')
-    const type = req.params.Type
-    const requesterName = req.headers['fspiop-source']
-    const requesterParticipantModel = await participant.validateParticipant(req.headers['fspiop-source'])
+    const type = params.Type
+    const requesterName = headers[Enums.Http.Headers.FSPIOP.SOURCE]
+    const requesterParticipantModel = await participant.validateParticipant(headers[Enums.Http.Headers.FSPIOP.SOURCE])
     if (requesterParticipantModel) {
-      if (Object.values(Enums.type).includes(type)) {
-        const response = await oracle.oracleRequest(req)
+      if (Object.values(Enums.Accounts.PartyAccountTypes).includes(type)) {
+        const response = await oracle.oracleRequest(headers, method, params, query)
         if (response && response.data && Array.isArray(response.data.partyList) && response.data.partyList.length > 0) {
           const options = {
             partyIdType: type,
-            partyIdentifier: req.params.ID
+            partyIdentifier: params.ID
           }
           const payload = {
             fspId: response.data.partyList[0].fspId
           }
-          await participant.sendRequest(req, requesterName, Enums.endpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_PUT, Enums.restMethods.PUT, payload, options)
+          await participant.sendRequest(headers, requesterName, Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_PUT, Enums.Http.RestMethods.PUT, payload, options)
         } else {
-          await participant.sendErrorToParticipant(req, requesterName, Enums.endpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_PUT_ERROR,
-            util.buildErrorObject(Errors.ErrorObject.PARTY_NOT_FOUND_ERROR, [{key: type, value: req.params.ID}]))
+          await participant.sendErrorToParticipant(requesterName, Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_PUT_ERROR,
+            ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.PARTY_NOT_FOUND).toApiErrorObject(), headers, params)
         }
       } else {
-        await participant.sendErrorToParticipant(req, requesterName, Enums.endpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_PUT_ERROR,
-          util.buildErrorObject(Errors.ErrorObject.ADD_PARTY_ERROR, [{key: type, value: req.params.ID}]))
+        await participant.sendErrorToParticipant(requesterName, Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_PUT_ERROR,
+          ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR).toApiErrorObject(), headers, params)
       }
       Logger.info('getParticipantsByTypeAndID::end')
     } else {
       Logger.error('Requester FSP not found')
-      // TODO: handle issue where requester fsp not found. Pass to error handling framework
+      throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found').toApiErrorObject()
```

Comment:
Not sure you you expect a JSON object to be thrown as an error :D

`throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found').toApiErrorObject()` --> `throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found')`